sonar-text-plugin/src/main/java/org/sonar/plugins/common/Analyzer.java

```diff
@@ -0,0 +1,144 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.common;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonarsource.analyzer.commons.ProgressReport;
+
+public class Analyzer {
+  private static final int REPORT_REFRESH_TIME_IN_SECONDS = 10;
+  private static final Logger LOG = LoggerFactory.getLogger(Analyzer.class);
+  private boolean displayHelpAboutExcludingBinaryFile = true;
+
+  private SensorContext sensorContext;
+  private List<Check> activeChecks;
+  private NotBinaryFilePredicate notBinaryFilePredicate;
+  private boolean analyzeAllFilesMode;
+
+  public Analyzer() {
+    // Empty analyzer by default, configuration provided by setters
+  }
+
+  public Analyzer sensorContext(SensorContext sensorContext) {
+    this.sensorContext = sensorContext;
+    return this;
+  }
+
+  public Analyzer activeChecks(List<Check> activeChecks) {
+    this.activeChecks = activeChecks;
+    return this;
+  }
+
+  public Analyzer notBinaryFilePredicate(NotBinaryFilePredicate notBinaryFilePredicate) {
+    this.notBinaryFilePredicate = notBinaryFilePredicate;
+    return this;
+  }
+
+  public Analyzer analyzeAllFilesMode(boolean analyzeAllFilesMode) {
+    this.analyzeAllFilesMode = analyzeAllFilesMode;
+    return this;
+  }
+
+  public void analyzeFiles(Collection<InputFile> inputFiles) {
+    List<String> filenames = inputFiles.stream().map(InputFile::toString).collect(Collectors.toList());
+    var progressReport = new ProgressReport("Progress of the text and secrets analysis", TimeUnit.SECONDS.toMillis(REPORT_REFRESH_TIME_IN_SECONDS));
+    progressReport.start(filenames);
+    var cancelled = false;
+
+    try {
+      for (InputFile inputFile : inputFiles) {
+        if (sensorContext.isCancelled()) {
+          cancelled = true;
+          break;
+        }
+
+        analyzeFile(inputFile);
+        progressReport.nextFile();
+      }
+    } finally {
+      if (cancelled) {
+        progressReport.cancel();
+      } else {
+        progressReport.stop();
+      }
+    }
+  }
+
+  private void analyzeFile(InputFile inputFile) {
+    try {
+      var inputFileContext = new InputFileContext(sensorContext, inputFile);
+      if (analyzeAllFilesMode) {
+        analyzeWithNotBinaryFileCheck(inputFileContext);
+      } else {
+        analyzeAllChecks(inputFileContext);
```

Comment:
This is not how the original logic does it. 

The `notBinaryFilePredicate` is applied everytime in the `analyze` method of the Sensor.
The `notBinaryFilePredicate` is not applied `Text...Sensor.getInputFiles` when `analyzeAllFiles` is true.

In your example the usage is entirely dependent on the `analyzeAllFiles`.

Reply:
I believe I did it on purpose.

When `analyzeAllFiles` is `true`, the `FilePredicate` is `notBinaryFilePredicate`, so we are in blacklist mode -> we analyze all files that are not detected as a binary file
Here the check on file content makes sense, because with this blacklist mode, we may still find binary files that were not detected by their extension.

When `analyzeAllFiles` is `false`, the `FilePredicate` is `or(textFilePredicate, LANGUAGE_FILE_PREDICATE)`, so we are in whitelist mode and will only analyze files that either:
- has an associated language
- has a text file extension (xml, txt, json, ...) as defined by the property `sonar.text.included.file.suffixes`

The check on the file content doesn't make much sense here, we are in whitelist mode, as per the above restrictions we shouldn't encounter a binary file.

We could still perform the check, but in that case, we should adapt our message and display it for every case it appears.
Instead of this:
```
'java' was added to the binary file filter because the file 'MyClass.java' is a binary file.
To remove the previous warning you can add the '.java' extension to the 'sonar.text.excluded.file.suffixes' property.
```
(wrong property)

We could instead have something like this:
```
The file 'MyClass.java' contains binary data and will not be analyzed.
Please check this file and/or remove the extension from the 'sonar.text.included.file.suffixes' property.
```
WDYT?

Reply:
Nice analysis of the context. I believe you're right.
I would like to keep this additional "safety net", and change the message as you have suggested.